src/routes/controllers/tag.controller.js

```diff
@@ -0,0 +1,104 @@
+import sequelize from '../../../.sequelizerc';
+import models from '../../db/models';
+import {
+  successResponse,
+  errorResponse,
+  verifyToken,
+} from '../../utils/helpers.utils';
+
+const { ArticleTags, Tags } = models;
+
+/**
+ * @description create tag for an article
+ * @param {*} tags
+ * @param {*} articleId
+ * @returns {boolean} response
+ */
+export async function createTag(tags, articleId) {
+  if (!Array.isArray(tags)) return null;
+  let createdTag = [];
+  for (let tag of tags) {
+    const response = await processTags(tag.toLowerCase(), articleId);
+    if (response.status === 'error') return response.message;
+    createdTag.push(response.message);
+  }
+  return createdTag;
+}
+
+/**
+ * @description update tag for an article
+ * @param {*} tags
+ * @param {*} articleid
+ * @returns {boolean} response
+ */
+export async function updateTag(tags, articleid) {
+  try {
+    if (!Array.isArray(tags)) return null;
+    for (let tag of tags) {
+      const removedTag = await removeTag(articleid);
+      if (removedTag.status !== 'deleted') throw new Error(removedTag.message);
+    }
+    const createdTag = await createTag(tags, articleid);
+    return createdTag;
+  } catch (error) {
+    return error.message;
+  }
+}
+
+/**
+ * @description delete a tag from article
+ * @param {string} articleId
+ * @returns {boolean} response
+ */
+export async function removeTag(articleId) {
+  console.log(articleId);
+  try {
+    const deletedTag = await ArticleTags.destroy({
+      where: { articleId },
+    });
+    return { status: 'deleted' };
```

Comment:
Is this function a controller endpoint? If it is, why did you choose to return status deleted instead of using the global response helper we use for uniformity?

Reply:
the function is not a controller endpoint. it's a function that **delete** a *tag* from the database.